src/services/filament/spool-manager.octoprint.service.ts

```diff
@@ -28,11 +28,15 @@ export class SpoolManagerOctoprintService implements FilamentPluginService {
   public getCurrentSpools(): Observable<Array<FilamentSpool>> {
     return this.callSpoolManagerAPI('hideInactiveSpools', 0, 3000, 'lastUse', 'desc').pipe(
       map((spools: SpoolManagerSpoolList): FilamentSpool[] => {
-        if (spools.selectedSpools[0]) {
-          return spools.selectedSpools.map(s => this.convertFilamentManagerSpool(s));
-        } else {
-          return null;
-        }
+        const sourceSpools = spools.selectedSpools?.some(spool => spool !== null)
+          ? spools.selectedSpools
+          : spools.allSpools?.slice(0, 5) || [];
```

Comment:
So if no spools are selected, this just assumes the first five overall spools are selected? If that's truly correct behavior, I think some explanation should be included as a comment. 

Additionally, I know the MMU plugin allows to change the capacity to accommodate modded MMUs, and I think the count here should respect that.